hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -174,6 +174,13 @@ public static OffsetRange[] computeOffsetRanges(Map<TopicPartition, Long> fromOf
           }
         }
       }
+      // We need to ensure every partition is part of returned offset ranges even if we are not consuming any new msgs (for instance, if its already caught up).
+      // as this will be tracked as the checkpoint, we need to ensure all partitions are part of final ranges.
+      fromOffsetMap.entrySet()
+          .stream()
+          .filter((kv) -> !finalRanges.containsKey(kv.getKey()))
+          .forEach((kv) -> finalRanges.put(kv.getKey(), Collections.singletonList(OffsetRange.create(kv.getKey(), kv.getValue(), kv.getValue()))));
```

Comment:
Generally, it's a bad pattern to filter and change the same map `finalRanges` in the Java stream operation.  You may collect the items into a list after filtering and then do foreach to avoid any unexpected behavior.

Reply:
+1

Reply:
Made the change to address this.